Feat(UI): Enhanced Theming and Component Refactoring

JianZcar opened the pull request:
Description

This pull request introduces a comprehensive theming system with light and dark modes, refactors several UI components to use daisyUI, and improves the overall user experience.

The primary motivation for these changes was to modernize the application's look and feel, improve code maintainability by leveraging a component-based approach with daisyUI, and introduce
a user-requested feature for theme selection.

Changes Made

 * Theming:
     * Defined custom light (default) and dark themes using daisyUI.
     * The dark theme is automatically enabled based on the user's prefers-color-scheme.
     * A theme switcher dropdown has been added to allow users to manually select a theme.
     * The primary color for both themes has been set to black to maintain brand consistency.

 * Component Refactoring:
     * The user receipt view has been refactored from a custom modal to a more accessible and standardized daisyUI modal (<dialog>).
     * The user purchase history is now displayed using a reusable card.php component.
     * The user QR code page has been updated to use the new theming system and daisyUI components.

 * UI Enhancements:
     * User purchase cards now have a hover effect to improve interactivity.
     * Unused CSS classes have been removed to streamline the stylesheet.

 * Dependency Updates:
     * daisyUI has been integrated as a Tailwind CSS plugin.

How to Test

 1. Navigate to the user dashboard.
 2. Verify that the new theme switcher dropdown is present and allows you to switch themes.
 3. Verify that the dark theme is automatically applied if your system is set to dark mode.
 4. Click on a purchase to view the receipt. Verify that it opens in a daisyUI modal.
 5. Navigate to the QR code page and verify that it is styled correctly with the new theme.
 6. Hover over purchase cards to ensure the hover effect is working as expected.

Commit:
feat: integrate DaisyUI and refactor user purchase history

Integrates DaisyUI as a Tailwind CSS plugin to enhance the user interface.

Key changes:
- Adds DaisyUI to the project dependencies.
- Refactors the user purchase history display into a reusable card component ('card.php').
- Creates a new receipt modal ('receipt_modal.php') for displaying purchase details.
- Updates the main user page ('index.php') to use the new card component.
- Includes project configuration for bun ('bun.lock') and nginx ('nginx.conf').

Commit:
feat(ui): add theme switcher and refactor receipt modal

Introduces a theme switcher dropdown powered by daisyUI, allowing users to select different visual themes.

- Refactors the user receipt view from a custom modal to a daisyUI modal component (<dialog>).
- Updates JavaScript to use the native showModal() method for the new dialog.
- Enhances the styling of user purchase cards with hover effects.

Commit:
feat(theme): define and apply default light/dark themes

- Defines custom `light` (default) and `dark` (prefers-color-scheme) themes.
- Sets the primary color to black for both new themes to ensure brand consistency.
- Updates the user QR code page to adopt the new theming, replacing old styles with daisyUI components and theme variables.
- Removes unused CSS classes to streamline the stylesheet.

FPPinedaJr reviewed (changes requested):
1. Include the following theme/s:

- https://www.canva.com/design/DAG2NLTo1H4/iFM1evmvCMJKlIVURRBLbw/edit
- https://www.canva.com/design/DAG2MmI1P_4/txS434fLHB0evr61LVEwwQ/edit
- https://www.canva.com/design/DAGfdc2vX1o/smYx3-oRrnMHchsPmqFWYg/edit

2. Change the theme select to be in another page (not in header)

3. Fix styling in Sign In [see picture below]

<img width="1920" height="953" alt="image" src="https://github.com/user-attachments/assets/5d57bd4f-8de9-44fb-bc35-d05878d1d66f" />


4. Add theme change feature to admin pages 

5. Fix the footer not anchored in the bottom [see picture below]

<img width="1920" height="953" alt="chrome_aUB5qBKJMT" src="https://github.com/user-attachments/assets/ca47dee3-99a4-41b6-8b2c-eb1cbc31d0da" />

6. Make it so the icon also changes color

7. Default theme is green (original)

8. Make sure that the theme is saved when changing page (also after logging out and resigning in)

Commit:
green default theme

Commit:
blue and red theme

Commit:
persistent theme

Commit:
persistent theme

Commit:
admin home

Commit:
admin home

Commit:
Merge branch 'theme' of github.com:masisikip/MoneyMo into theme

Commit:
Merge branch 'theme' of github.com:masisikip/MoneyMo into theme

Commit:
Merge branch 'theme' of github.com:masisikip/MoneyMo into theme

Commit:
idk

Commit:
feat(theme): Implement consistent theming across admin pages

Replaces hardcoded colors with theme variables (bg-primary, bg-base-200, text-primary) for a more consistent look and feel.

- Updates buttons, backgrounds, and text colors in the inventory and statistics pages.
- Modifies the chart in the statistics page to generate colors based on the primary theme color, improving visual consistency.
- Adds new hover utility classes for theme colors.
- Updates sidebar navigation links to use theme-based hover effects.

Commit:
theming on statistics page

Commit:
feat(theme): Apply theme across admin and user pages

This commit continues the theming effort by applying the consistent color scheme to several pages within the admin and user sections.

- Replaces hardcoded colors with theme variables (e.g., 'bg-primary', 'bg-base-200') on the following pages:
  - admin/code
  - admin/export
  - admin/import
  - admin/item
  - admin/scanner
  - admin/users
- Includes the 'theme.php' file to enable theme switching on these pages.
- Updates the theme dropdown on the user profile page to be full-width for better usability.
- Adds new utility classes to 'styles.css' to support the theme changes.

Commit:
refactor(theme): Simplify theme definition and replace logo with SVG

This commit refactors the theme implementation and replaces the logo image with an inline SVG for better performance and scalability.

- Replaces the logo image with an inline SVG, colored with the primary theme color.
- Simplifies the CSS by removing unused classes and refactoring the theme definitions.
- Removes the explicit light theme from 'input.css', relying on the default configuration.
